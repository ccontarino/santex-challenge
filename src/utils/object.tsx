export const deepCopyArray = (array: any[]) =>
  JSON.parse(JSON.stringify(array));
