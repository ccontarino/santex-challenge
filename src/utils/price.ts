export function formatPrice(number: string | number) {
  const price = Number(number);
  var formattedNumber = number.toLocaleString('en-US', { style: 'decimal', maximumFractionDigits: 0 });
  return formattedNumber.replace(".", ",");
}
