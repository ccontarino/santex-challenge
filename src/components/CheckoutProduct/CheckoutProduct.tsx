import * as React from 'react';
import styled from 'styled-components';
import { Order, ProductVariant } from '../../types/types';
import { formatPrice } from '../../utils/price';

interface componentNameProps {
  index?: number;
  name?: string;
  productImage?: string | undefined;
  productDescription?: string;
  productTitle?: string;
  id: string | undefined;
  variant?: ProductVariant;
  order?: Order;
}

const CheckoutProductContainer = styled.div`
  background-color: #fff;
  display: flex;
  margin: 0 0 10px 0;
  width: 100%;
  height: 120px;
  border-radius: 8px;
  border: 1px solid #ccc;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
`;
const ProductImage = styled.img`
  height: 100%;
  padding: 10px;
  flex: 1;
  border-radius: 8px;
  object-fit: cover;
`;
const VerticalLine = styled.div`
  border-left: 1px solid #ccc;
  height: 100%;
  width: 3px;
`;
const TitleContainer = styled.div`
  padding: 8px;
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
`;

const CheckoutProduct = ({
  productImage,
  name,
  variant,
  order,
}: componentNameProps) => {
  console.log('variant :: ', variant);
  console.log('variant :: ', variant);
  return (
    <CheckoutProductContainer>
      <ProductImage src={productImage} loading="lazy" />
      <VerticalLine />
      <TitleContainer>
        <div>{name}</div>
        <span>
          <div>Variant:{variant?.name}</div>
        </span>
        <div>Price:{formatPrice(variant?.price)}</div>
      </TitleContainer>
    </CheckoutProductContainer>
  );
};

export default CheckoutProduct;
