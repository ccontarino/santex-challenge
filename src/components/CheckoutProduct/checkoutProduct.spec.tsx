import React from 'react';
import { render, screen } from '@testing-library/react';
import CheckoutProduct from './checkoutProduct';
import { formatPrice } from '../../utils/price';

describe('CheckoutProduct', () => {
  it('renders product  name, variant name, and price', () => {
    const productImage = 'image.jpg';
    const name = 'Product Name';
    const variant = { name: 'Variant Name', price: 9.99, id: '123' };
    const checkoutProduct = render(
      <CheckoutProduct
        productImage={productImage}
        name={name}
        variant={variant}
      />
    );
    expect(checkoutProduct).toMatchSnapshot();
    expect(screen.getByText(name)).toBeInTheDocument();
    expect(screen.getByText(variant.name)).toBeInTheDocument();
    expect(
      screen.getByText(`${formatPrice(variant?.price)}`)
    ).toBeInTheDocument();
  });
});
