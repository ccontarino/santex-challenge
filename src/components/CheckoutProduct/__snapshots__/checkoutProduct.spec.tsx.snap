// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutProduct renders product  name, variant name, and price 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-aXZVg bBwTRW"
      >
        <img
          class="sc-gEvEer ejGPNF"
          loading="lazy"
          src="image.jpg"
        />
        <div
          class="sc-eqUAAy ibJorR"
        />
        <div
          class="sc-fqkvVR fUlCrx"
        >
          <div
            class="sc-dcJsrY dvLWId"
          >
            Product Name
          </div>
          <div
            class="sc-dcJsrY dvLWId"
          >
            Variant Name
          </div>
          <div
            class="sc-dcJsrY dvLWId"
          >
            <span>
              Price:
            </span>
            $9.99
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-aXZVg bBwTRW"
    >
      <img
        class="sc-gEvEer ejGPNF"
        loading="lazy"
        src="image.jpg"
      />
      <div
        class="sc-eqUAAy ibJorR"
      />
      <div
        class="sc-fqkvVR fUlCrx"
      >
        <div
          class="sc-dcJsrY dvLWId"
        >
          Product Name
        </div>
        <div
          class="sc-dcJsrY dvLWId"
        >
          Variant Name
        </div>
        <div
          class="sc-dcJsrY dvLWId"
        >
          <span>
            Price:
          </span>
          $9.99
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
